### DISCUSSION TITLE
Decision for dependencies and devdependencies

### DISCUSSION THREAD

**Discussion by YaminoYuki** [webpack/webpack#16113] [Q&A] [answered]
From this:
https://jsramblings.com/do-dependencies-devdependencies-matter-when-using-webpack/
It seems that when we use webpack it doesn't matter where to put our packages, either in dependencies or devdependencies it will be the same, webpack will bundle what is needed.
But for licenses issue, I want to know the exact place where to put each needed package , so the things that are not bundled I will add them in devdependencies.
I am thinking to make my decision upon theses rules:
1- Use the result from : https://github.com/webpack-contrib/webpack-bundle-analyzer each packages is in the stat will be treated as a dependency.
2- Use the npm convention, when they do not use --dev for npm install, it will be treated as a dependency.
3- Any module where we had to "import" it or "require" it, will be considered as a dependency.

Do you think that is enough ? also what about the dependencies of some of my devdependencies? for example babel/runtime is bundled (webpack bundle analyzer shows that) but this module is not mentioned in my package.json , it is mentioned as a dependency for a devdependency

**Comment by vankop** [comment, marked as answer]
for webpack it doesn't matter.. technically you can use your plan..

> also what about the dependencies of some of my devdependencies?

no, you don't need to add them.

**Reply by YaminoYuki** [reply]
Thank you very much. it seems I was mistaken regarding babel/runtime I found one of the dependencies which uses it as a dependency
